tests/robustness/traffic/kubernetes.go

```diff
@@ -151,13 +152,19 @@ func (t kubernetesTraffic) Write(ctx context.Context, kc *kubernetesClient, ids
 		if rev == 0 {
 			return errors.New("storage empty")
 		}
-		if count > t.averageKeyCount*3/2 {
+		if count > t.averageKeyCount*3/2 && nonUniqueWriteLimiter.Take() {
 			_, err = kc.OptimisticDelete(writeCtx, key, rev)
+			nonUniqueWriteLimiter.Return()
 		} else {
-			op := pickRandom(t.writeChoices)
+			choices := t.writeChoices
+			if !nonUniqueWriteLimiter.Take() {
+				choices = filterOutNonUniqueKuberntesWrites(t.writeChoices)
+			}
+			op := pickRandom(choices)
 			switch op {
 			case KubernetesDelete:
 				_, err = kc.OptimisticDelete(writeCtx, key, rev)
+				nonUniqueWriteLimiter.Return()
```

Comment:
Not sure I am understand the code correctly: If the number of client (8) is larger than non-unique write (3), if 5 clients don't get true from `Take()` at Line 160 at the same time, is it safe to call the `Return` here?

Reply:
If they got `Take() = false`, then `op` should never be equal to `KubernetesDelete`.